pkg/frontend/admin_openshiftcluster_serialconsole_test.go

```diff
@@ -0,0 +1,142 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	mock_adminactions "github.com/Azure/ARO-RP/pkg/util/mocks/adminactions"
+	testdatabase "github.com/Azure/ARO-RP/test/database"
+)
+
+const (
+	mockSubID    = "00000000-0000-0000-0000-000000000000"
+	mockTenantID = "00000000-0000-0000-0000-000000000000"
+)
+
+func databaseFixture(f *testdatabase.Fixture) {
+	f.AddOpenShiftClusterDocuments(&api.OpenShiftClusterDocument{
+		Key: strings.ToLower(testdatabase.GetResourcePath(mockSubID, "resourceName")),
+		OpenShiftCluster: &api.OpenShiftCluster{
+			ID: strings.ToLower(testdatabase.GetResourcePath(mockSubID, "resourceName")),
+			Properties: api.OpenShiftClusterProperties{
+				ClusterProfile: api.ClusterProfile{
+					ResourceGroupID: fmt.Sprintf("/subscriptions/%s/resourceGroups/test-cluster", mockSubID),
+				},
+			},
+		},
+	})
+
+	f.AddSubscriptionDocuments(&api.SubscriptionDocument{
+		ID: mockSubID,
+		Subscription: &api.Subscription{
+			State: api.SubscriptionStateRegistered,
+			Properties: &api.SubscriptionProperties{
+				TenantID: mockTenantID,
+			},
+		},
+	})
+}
+
+func TestGetAdminOpenShiftClusterSerialConsole(t *testing.T) {
+	tests := []struct {
+		name           string
+		vmName         string
+		resourceID     string
+		fixture        func(*testdatabase.Fixture)
+		mocks          func(*mock_adminactions.MockAzureActions)
+		wantStatusCode int
+		wantError      string
+	}{
+		{
+			name:       "valid request returns console output",
+			vmName:     "master-0",
+			resourceID: strings.ToLower(testdatabase.GetResourcePath(mockSubID, "resourceName")),
```

Comment:
Thought for a future PR:

I wonder could we neaten up all the URL Parameter Keys to a single constants file? Maybe this could even be injected into the Frontend.

Reply:
A Jira card similar to the one that drove this PR would be fine.